src/integrationTest/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTaskTestForFailedStatus.java

```diff
@@ -0,0 +1,170 @@
+    @After
+    public void cleanUp() throws Exception {
+        testContainer.deleteIfExists();
+        envelopeRepository.deleteAll();
+        envelopeStateRepository.deleteAll();
+    }
+
+    @Test
+    public void should_record_failure_of_upload_when_document_management_returns_empty_response() throws IOException {
+        // given
+        given(documentManagementService.uploadDocuments(getUploadResources())).willReturn(Collections.emptyMap());
+
+        // when
+        blobProcessorTask.processBlobs();
+
+        // then
+        Envelope actualEnvelope = envelopeRepository.findAll().get(0);
+
+        assertThat(actualEnvelope.getScannableItems()).extracting("documentUrl").allMatch(ObjectUtils::isEmpty);
+
+        // and
+        List<EnvelopeState> envelopeStates = envelopeStateRepository.findAll();
+        assertThat(envelopeStates).hasSize(1);
+
+        EnvelopeState envelopeState = envelopeStates.get(0);
+        assertThat(envelopeState)
+            .extracting("container", "zipFileName", "status")
+            .hasSameElementsAs(ImmutableList.of(testContainer.getName(), ZIP_FILE_NAME, DOC_UPLOAD_FAILURE));
+        assertThat(envelopeState.getId()).isNotNull();
+        assertThat(envelopeState.getReason()).isNotBlank();
+        assertThat(envelopeState.getEnvelope().getId()).isEqualTo(actualEnvelope.getId());
+    }
+
+    @Test
+    public void should_record_failure_of_upload_when_document_management_throws_exception() throws IOException {
+        // given
+        Throwable throwable = new UnableToUploadDocumentException("oh no", null);
+        given(documentManagementService.uploadDocuments(getUploadResources())).willThrow(throwable);
+
+        // when
+        blobProcessorTask.processBlobs();
+
+        // then
+        Envelope actualEnvelope = envelopeRepository.findAll().get(0);
+
+        assertThat(actualEnvelope.getScannableItems()).extracting("documentUrl").allMatch(ObjectUtils::isEmpty);
+
+        // and
+        List<EnvelopeState> envelopeStates = envelopeStateRepository.findAll();
+        assertThat(envelopeStates).hasSize(1);
+
+        EnvelopeState envelopeState = envelopeStates.get(0);
+        assertThat(envelopeState)
+            .extracting("container", "zipFileName", "status")
+            .hasSameElementsAs(ImmutableList.of(testContainer.getName(), ZIP_FILE_NAME, DOC_UPLOAD_FAILURE));
+        assertThat(envelopeState.getId()).isNotNull();
+        assertThat(envelopeState.getReason()).isEqualTo(throwable.getMessage());
+        assertThat(envelopeState.getEnvelope().getId()).isEqualTo(actualEnvelope.getId());
+    }
+
+    private void uploadZipToBlobStore() throws Exception {
+        byte[] zipFile = toByteArray(getResource(ZIP_FILE_NAME));
+
+        CloudBlockBlob blockBlobReference = testContainer.getBlockBlobReference(ZIP_FILE_NAME);
+        blockBlobReference.uploadFromByteArray(zipFile, 0, zipFile.length);
+    }
+
+    private List<Pdf> getUploadResources() throws IOException {
```

Comment:
I think method like these could be extracted to a common base class instead of repeating it in each class ? Could be a separate PR for refactoring though.

Reply:
True. best to leave for later once all the work around it is done and you are able to see bigger picture. This function will also change in the next PR 😉